crates/e2e/tests/e2e/services.rs

```diff
@@ -175,6 +175,67 @@ pub async fn deploy_mintable_token(web3: &Web3) -> ERC20Mintable {
         .expect("MintableERC20 deployment failed")
 }
 
+pub struct WethPoolConfig {
+    pub token_amount: U256,
+    pub weth_amount: U256,
+}
+
+pub struct MintableToken {
+    pub contract: ERC20Mintable,
+}
+
```

Comment:
ubernit; Maybe this type should be in its own module? So the `services` is just for setting up the service components and then another module for on-chain components (like Uniswap and Balancer pools?).

Reply:
Right, it's there right now because it's the same place where we create a mintable token instance.
I see that this plus the suggestion of using `deploy_token_with_weth_uniswap_pool` in other tests include a lot of (mostly trivial) changes, so I plan to both in another PR and merge this as it is.

Reply:
This is https://github.com/cowprotocol/services/pull/818. We can continue the discussion there if needed.